packages/@aws-cdk/aws-scheduler-targets-alpha/lib/lambda-invoke.ts

```diff
@@ -1,34 +1,23 @@
 import { ISchedule, IScheduleTarget } from '@aws-cdk/aws-scheduler-alpha';
-import { Names } from 'aws-cdk-lib';
 import { IRole } from 'aws-cdk-lib/aws-iam';
 import * as lambda from 'aws-cdk-lib/aws-lambda';
 import { ScheduleTargetBase, ScheduleTargetBaseProps } from './target';
-import { sameEnvDimension } from './util';
 
 /**
  * Use an AWS Lambda function as a target for AWS EventBridge Scheduler.
  */
 export class LambdaInvoke extends ScheduleTargetBase implements IScheduleTarget {
+  private readonly func: lambda.IFunction;
+
   constructor(
-    private readonly func: lambda.IFunction,
-    private readonly props: ScheduleTargetBaseProps,
+    func: lambda.IFunction,
+    props: ScheduleTargetBaseProps,
   ) {
     super(props, func.functionArn);
+    this.func = func;
```

Comment:
why is this needed?

Reply:
So because `props` is no longer used in `addTargetActionToRole`, TS was complaining:
```
TS6138: Property props is declared but its value is never read
```
To fix, it could be:
```ts
constructor(
    private readonly func: lambda.IFunction,
    props: ScheduleTargetBaseProps,
  ) {
    super(props, func.functionArn);
  }
```
But I do not want to mix the 2 style. So went with what it is now to make the code more consistent.

Reply:
oh I see, I thought `func` is not used either so I asked this question, but I missd that it's still used at the end `this.func.grantInvoke(role);`.

Reply:
Will this apply to other target types as well? If so we should document as TODO to keep the code consistent

Reply:
Yes, same will apply to the other targets if they have the same change. But since it is a code styling issue (no functional difference), we can update the other targets as they get graduated.